lib/driver.rb

```diff
@@ -19,29 +22,77 @@ def initialize(input)
       @status = input[:status] == nil ? :AVAILABLE : input[:status]
 
       @trips = input[:trips] == nil ? [] : input[:trips]
-    end
+    end # initialize
+
+    def completed_trips
+      @trips.find_all {|trip| trip.end_time != nil}
+    end # completed_trips
```

Comment:
I love that you've broken this out as a separate helper method! This makes the other code in this file much easier to read.